server/routes/functionalSkillsResolver.ts

```diff
@@ -13,7 +13,7 @@ import dateComparator from './dateComparator'
 const mostRecentFunctionalSkills = (allFunctionalSkills: FunctionalSkills): FunctionalSkills => {
   return {
     ...allFunctionalSkills,
-    assessments: mostRecentAssessments(allFunctionalSkills.assessments || []),
+    assessments: mostRecentAssessments([...(allFunctionalSkills.assessments || [])]),
```

Comment:
As expected, the tests in the controller proved that we need to destructure the arrays before sorting and filtering in each of these method calls. 
Without destructuring we end up mutating the source object! 😬